web-components/chart/chart.tsx

```diff
@@ -0,0 +1,309 @@
+      // Vega (pie charts), and we only compile the spec if we're using VegaLite.
+      compiledSpec = this.vegaLite
+        ? VegaLite.compile(this.config).spec
+        : this.config;
+    } else {
+      // If we are using a selection, we get an array of the unique values in the
+      // chosen field to use to populate the dropwdown menu.
+      this.selectField = this.config.selection.select.fields[0];
+      this.selectOptions = [
+        ...new Set(this.dataset.map(item => item[this.selectField])),
+      ].sort();
+
+      // We update the config to reflect the unique values we just found
+      this.config.selection.select.bind.options = this.selectOptions;
+
+      // After updating the config, we compile it to Vega and parse it
+      compiledSpec = VegaLite.compile(this.config).spec;
+
+      /* By default Vega adds and event listener to our selection that
+          fires when the chart is clicked. The click sets the selection
+          back to null, so the user's selection gets undone.
+
+          To get around this, we remove the event listener configuration
+          from the compiled Vega spec.
+        */
+      const selectSignal = compiledSpec.signals.find(
+        elem => elem.name === `select_${this.selectField}`
+      );
+      delete selectSignal.on;
+    }
+    // After updating the config and compiling if necessary, we initialize a
+    // Vega view object.
+    this.view = new Vega.View(Vega.parse(compiledSpec)).renderer('svg');
+  }
+
+  componentDidLoad() {
+    // Once the component loads, we initialize the view.
+    this.view.initialize(`#${this.chartID}`);
+    // Attach tooltips to the chart
+    vegaTooltip(this.view);
+    // If we are using selections, we build the dropdown.
+    if (this.config.selection) {
+      this.buildDropDownSelection();
+    }
+    // On load, Vega creates the chart svg and a div for any associated selections.
+    // We want to wrap just the chart svg it creates inside of a div so we can control
+    // horizontal scrolling for only the chart and not any elements around or below.
+    const chartWrapper = document.createElement('div');
+    chartWrapper.className = 'cob-chart-wrapper';
+    // Get the chart svg element and its parent node
+    const chartSVG = this.el.querySelector('svg');
+    const svgParent = chartSVG.parentNode;
+    // Update the children of the parent node
+    svgParent.replaceChild(chartWrapper, chartSVG);
+    chartWrapper.appendChild(chartSVG);
+    // Lastly, we set the chart width on load so we can
+    // make sure it fits nicely into the page.
+    this.setChartWidth();
+  }
+
+  // We custom build the dropdown selection to adhere to our styles and branding by
+  // maniupulating the div Vega creates for selections.
+  // This currently supports only one selection that is a dropdown/select element.
+  buildDropDownSelection() {
+    // We grab the div holding the chart and selection elements.
+    const selectDiv = this.el.querySelector('div');
```

Comment:
I ended up using the chart id so this line now reads: `const selectDiv = this.el.querySelector('div#${this.chartID}');` since I thought that would be even safer than a class name. Can you confirm that was a smart move?

Reply:
Yep, that's fine. You don't even need the `div` at that point and could use `getElementById`